test/safe.test.js

```diff
@@ -1,377 +1,190 @@
-
-  beforeAll(() => {
-    accounts = new Array(4).fill({}).map((item, index) => {
-      return getAccount(index);
-    });
-
-    core = createCore();
-  });
+  const core = createCore();
 
   describe('when a new Safe gets manually created', () => {
+    const nonce = generateSaltNonce();
     let safeAddress;
-    let nonce;
 
-    beforeAll(async () => {
-      nonce = Date.now();
-      safeAddress = await core.safe.prepareDeploy(accounts[0], {
-        nonce,
-      });
-    });
+    it('should deploy a Safe successfully', async () => {
+      safeAddress = await core.safe.deploySafe(accounts[0], { nonce });
 
-    it('should be a valid address', () => {
       expect(web3.utils.isAddress(safeAddress)).toBe(true);
-    });
-
-    it('should have predicted its future Safe address', async () => {
-      const predictedSafeAddress = await core.safe.predictAddress(accounts[0], {
-        nonce,
-      });
 
-      expect(web3.utils.isAddress(predictedSafeAddress)).toBe(true);
-      expect(predictedSafeAddress).toBe(safeAddress);
+      return core.safe
+        .isDeployed(accounts[0], { safeAddress })
+        .then((isDeployed) => expect(isDeployed).toBe(true));
     });
 
-    it('should return the correct status', async () => {
-      const status = await core.safe.getSafeStatus(accounts[0], {
-        safeAddress,
-      });
-
-      expect(status.isCreated).toBe(true);
-      expect(status.isDeployed).toBe(false);
-    });
-
-    it('should recover the same Safe address when doing it again', async () => {
-      const safeAddressAgain = await core.safe.prepareDeploy(accounts[0], {
-        nonce,
-      });
-
-      expect(safeAddressAgain).toBe(safeAddress);
-    });
-
-    it('should be able to manually fund it for deployment', async () => {
-      expect(
-        await core.safe.isFunded(accounts[0], {
-          safeAddress,
-        }),
-      ).toBe(false);
-
-      await fundSafe(accounts[0], safeAddress);
-
-      expect(
-        await core.safe.isFunded(accounts[0], {
-          safeAddress,
-        }),
-      ).toBe(true);
-
-      const result = await core.safe.deploy(accounts[0], {
-        safeAddress,
-      });
+    it('should get the last version of Safe Contract by default', () =>
+      core.safe
+        .getVersion(accounts[0], { safeAddress })
+        .then((version) => expect(version).toBe(SAFE_LAST_VERSION)));
 
-      // .. wait for Relayer to really deploy Safe
-      await core.utils.loop(
-        () => web3.eth.getCode(safeAddress),
-        isContractDeployed,
-        {
-          label: 'Wait until Safe got deployed',
-        },
-      );
+    it('should get the safe address of the owner', () =>
+      core.safe
+        .getAddresses(accounts[0], {
+          ownerAddress: accounts[0].address,
+        })
+        .then((safeAddresses) => expect(safeAddresses).toContain(safeAddress)));
 
-      // Deploy Token as well to pay our fees later
-      await core.token.deploy(accounts[0], {
```

Comment:
where are we doing the token deployment?

Reply:
Nowhere, it was something not needed for these tests.